### DISCUSSION TITLE
useMatches hook like Remix

### DISCUSSION THREAD

**Discussion by manyu007vit** [vercel/next.js#46815] [Help]
### Summary

Can we expect useMatches kind of hook in NextJS like the one part of Remix framework?

**Comment by CanRau** [comment]
Yes would also love something like this in conjunction with something like [handle](http://remix.run/docs/en/main/route/handle) more details in the [useMatches](http://remix.run/docs/en/main/hooks/use-matches) docs.

Looking for the best way to be able to tell the header navigation and footer on what backgrounds they're rendered on but from each page itself. Maybe I could re-structure my design, but this is one use-case I think this would be useful 👍🏼
